src/services/navigationBar.ts

```diff
@@ -269,21 +288,70 @@ namespace ts.NavigationBar {
                 addLeafNode(node);
                 break;
 
+            case SyntaxKind.CallExpression:
             case SyntaxKind.BinaryExpression: {
                 const special = getAssignmentDeclarationKind(node as BinaryExpression);
                 switch (special) {
                     case AssignmentDeclarationKind.ExportsProperty:
                     case AssignmentDeclarationKind.ModuleExports:
-                    case AssignmentDeclarationKind.PrototypeProperty:
-                    case AssignmentDeclarationKind.Prototype:
                         addNodeWithRecursiveChild(node, (node as BinaryExpression).right);
                         return;
+                    case AssignmentDeclarationKind.PrototypeProperty:
+                    case AssignmentDeclarationKind.Prototype: {
+                        const binaryExpression = (node as BinaryExpression);
+                        const assignmentTarget = binaryExpression.left as PropertyAccessExpression;
+                        const prototypeAccess = assignmentTarget.expression as PropertyAccessExpression;
+                        // If we see a prototype assignment, start tracking the target as a class
+                        if (isIdentifier(prototypeAccess.expression)) {

```

Comment:
```ts
foo = function () { };
foo.prototype = {};
```

This ends up being undefined in the above case and breaks badly enough to drop the whole nav tree.

Reply:
Fixed it.